cuqi/randomvariable/_orderedset.py

```diff
@@ -0,0 +1,59 @@
+class OrderedSet:
+    """A set (i.e. unique elements) that keeps its elements in the order they were added.
+
+    This is a minimal implementation of an ordered set, using a dictionary for storage.
+    """
```

Comment:
Yes - as you wrote its always None with item stored as key. I am sure there could be other ways to do this, but since its an internal structure within the RV this approach is OK for me.